javascript/javascript1/week2/StudentManager.js

```diff
@@ -0,0 +1,34 @@
+const class07Students = [];
+function addStudentToClass(studentName) {
+  if (studentName == "The Queen") {
+    class07Students.push(studentName);
+  } else if (studentName == "") {
+        console.log("Name must be filled out");
+  } else if (
+    class07Students.length < 6 &&
+    class07Students.includes(studentName) === false
+  ) {
+    class07Students.push(studentName);
+  } else if (class07Students.includes(studentName) === true) {
+    console.log("Student " + studentName + " is already in the class");
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+}
+
+addStudentToClass("Jon");
+addStudentToClass("Ben");
+addStudentToClass("Patricia");
+addStudentToClass("Bob");
+addStudentToClass("Susan");
+addStudentToClass("Susan");
+addStudentToClass("Matthew");
+addStudentToClass("The Queen");
+
+console.log(class07Students);
+
+function getNumberOfStudents() {
+    console.log(class07Students.length);
+}
+
+getNumberOfStudents();
```

Comment:
The Queen can be added to the class multiple times. Even if she is the Queen, she doesn't get that much of special treatment :).  The art of else if statements is making sure you catch all corner cases!